src/snowflake/cli/_plugins/streamlit/commands.py

```diff
@@ -200,14 +200,18 @@ def _migrate_v1_streamlit_to_v2(pd: ProjectDefinition):
     if pd.streamlit.additional_source_files:
         artifacts.extend(pd.streamlit.additional_source_files)
 
+    identifier = {"name": pd.streamlit.name}
+    if pd.streamlit.schema_name:
+        identifier["schema"] = pd.streamlit.schema_name
+    if pd.streamlit.database:
+        identifier["database"] = pd.streamlit.database
+
     data = {
         "definition_version": "2",
         "entities": {
-            "streamlit_app": {
+            pd.streamlit.name: {
```

Comment:
should we keep this simple to avoid any unintended consequences for special characters?

Reply:
This won't cause any issues because we are building dictionary that's feed to pydantic model. This is a value not dot-separated string.